### DISCUSSION TITLE
Facing error in Deploy/dev/Kubernetes/BootstrapMinimal

### DISCUSSION THREAD

**Discussion by Pawan33333** [Sunbird-Ed/Community#154] [General Discussion]
Started by user [Diksha POC](http://35.223.161.5:8080/user/dikshapoc)
Checking out git https://github.com/project-sunbird/sunbird-devops.git into /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/BootstrapMinimal@script/fdfb149ced34077fca57da60c0df0dd05ff3d2c9d26fff4d9502ef43db314dd2 to read kubernetes/pipelines/bootstrap_minimal/Jenkinsfile
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/BootstrapMinimal@script/fdfb149ced34077fca57da60c0df0dd05ff3d2c9d26fff4d9502ef43db314dd2/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-devops.git # timeout=10
Using shallow fetch with depth 1
Fetching upstream changes from https://github.com/project-sunbird/sunbird-devops.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --tags --progress --depth=1 -- https://github.com/project-sunbird/sunbird-devops.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse origin/release-4.6.0_RC4^{commit} # timeout=10
 > git rev-parse release-4.6.0_RC4^{commit} # timeout=10
Checking out Revision 298c0e0cb57f5d8fa32bbc3676ae57c6c2127a6f (release-4.6.0_RC4)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 298c0e0cb57f5d8fa32bbc3676ae57c6c2127a6f # timeout=10
Commit message: "Issue #SB-28497 feat:updated enrollment list size env (#3180)"
First time build. Skipping changelog.
Loading library deploy-conf@heads/shared-lib
Attempting to resolve heads/shared-lib from remote references...
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git ls-remote -- https://github.com/project-sunbird/sunbird-devops.git # timeout=10
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/BootstrapMinimal@libs/4d80a6c970e5bba7406a7e32771e5156e33649aa2faaf628a0193bf413a1701e/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-devops.git # timeout=10
Using shallow fetch with depth 1
Fetching upstream changes from https://github.com/project-sunbird/sunbird-devops.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --no-tags --progress --depth=1 -- https://github.com/project-sunbird/sunbird-devops.git +refs/heads/*:refs/remotes/origin/* refs/heads/shared-lib # timeout=10
Checking out Revision 778da7ce7f97efd003eed98071921a837d209343 (778da7ce7f97efd003eed98071921a837d209343)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 778da7ce7f97efd003eed98071921a837d209343 # timeout=10
Commit message: "fix: move variables to be accessible across both try blocks"
 > git rev-list --no-walk 778da7ce7f97efd003eed98071921a837d209343 # timeout=10
[Pipeline] Start of Pipeline
[Pipeline] node
Running on [Jenkins](http://35.223.161.5:8080/computer/(built-in)/) in /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/BootstrapMinimal
[Pipeline] {
[Pipeline] stage
[Pipeline] { (checkout public repo)
[Pipeline] echo
Found .git folder. Clearing it..
[Pipeline] sh
+ git clean -fxd
Skipping repository private/
Removing private@tmp/
[Pipeline] checkout
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/BootstrapMinimal/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-devops.git # timeout=10
Using shallow fetch with depth 1
Fetching upstream changes from https://github.com/project-sunbird/sunbird-devops.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --tags --progress --depth=1 -- https://github.com/project-sunbird/sunbird-devops.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse origin/release-4.6.0_RC4^{commit} # timeout=10
 > git rev-parse release-4.6.0_RC4^{commit} # timeout=10
Checking out Revision 298c0e0cb57f5d8fa32bbc3676ae57c6c2127a6f (release-4.6.0_RC4)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 298c0e0cb57f5d8fa32bbc3676ae57c6c2127a6f # timeout=10
Commit message: "Issue #SB-28497 feat:updated enrollment list size env (#3180)"
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (deploy)
[Pipeline] sh
+ echo Deploy/dev/Kubernetes/BootstrapMinimal
[Pipeline] sh
+ echo Deploy/dev/Kubernetes/BootstrapMinimal
[Pipeline] sh
+ echo Deploy/dev/Kubernetes/BootstrapMinimal
[Pipeline] sh
+ pwd
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (ansible-run)
[Pipeline] echo
{currentWs=/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/BootstrapMinimal, env=dev, module=Kubernetes, ansiblePlaybook=/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/BootstrapMinimal/kubernetes/ansible/bootstrap_minimal.yaml, ansibleExtraArgs=--vault-password-file /var/lib/jenkins/secrets/vault-pass -v}
[Pipeline] dir
Running in /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/BootstrapMinimal/private
[Pipeline] {
[Pipeline] echo
Info: Branch override is disabled
[Pipeline] checkout
The recommended git tool is: NONE
using credential github-cred
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/BootstrapMinimal/private/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/dikshauat/Diksh-POC.git # timeout=10
Using shallow fetch with depth 1
Fetching upstream changes from https://github.com/dikshauat/Diksh-POC.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
using GIT_ASKPASS to set credentials 
 > git fetch --no-tags --progress --depth=1 -- https://github.com/dikshauat/Diksh-POC.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse origin/release-4.3.0^{commit} # timeout=10
Checking out Revision 1e14fcd45b4de7aceac993cccc1d1151b81343b0 (origin/release-4.3.0)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 1e14fcd45b4de7aceac993cccc1d1151b81343b0 # timeout=10
Commit message: "Update common.yml"
 > git rev-list --no-walk 447f68006dce12c924dda092310bd2e2773d3cdc # timeout=10
[Pipeline] }
[Pipeline] // dir
[Pipeline] sh
+ rsync -Lkr /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/BootstrapMinimal/private/ansible/inventory/dev/Kubernetes/* /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/BootstrapMinimal/ansible/inventory/env/
rsync: link_stat "/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/BootstrapMinimal/private/ansible/inventory/dev/Kubernetes/*" failed: No such file or directory (2)
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1196) [sender=3.1.2]
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // stage
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (slack_notify)
[Pipeline] sh
+ git rev-parse --short HEAD
[Pipeline] sh
+ rev
+ cut -d / -f1
+ git name-rev --name-only HEAD
+ rev
[Pipeline] sh
+ echo Deploy/dev/Kubernetes/BootstrapMinimal
[Pipeline] echo
Could not find env specific Slack channel. Check for global slack channel..
[Pipeline] echo
Could not find slack environment variable. Skipping slack notification..
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (email_notify)
[Pipeline] sh
+ echo Deploy/dev/Kubernetes/BootstrapMinimal
[Pipeline] echo
Could not find env specific email group. Check for global email group..
[Pipeline] echo
Could not find global email group variable. Skipping email notification..
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: script returned exit code 23
Finished: FAILURE

**Comment by beepdot** [comment]
Hello there.. Good day!
Do you have this directory in your private repo? `ansible/inventory/dev/Kubernetes`

**Reply by Pawan33333** [reply]
under this directory - ansible/inventory/dev/ my common, hosts and secrets files are there of core, DP and KP and yes it is in private repo

**Reply by beepdot** [reply]
There should be a `Kubernetes` directory also. Please see the template here - https://github.com/project-sunbird/sunbird-devops/tree/master/private_repo/ansible/inventory/dev

**Reply by Pawan33333** [reply]
Thanks a lot its working
